File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(x, y):
    return [i for i in range(x, y + 1)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc, s, x ,y):
    if s=='s':
        return sum(inpFunc(x , y))
    else:
        return sum(inpFunc(x , y))/(y - x + 1)

print(applicatorFunc(listFunc, 's', 1 , 10))

File: lab2-functionalprimer/javascript/func_as_arg.js
//Create list of ints from 1 to 5, Haskell equivalent [1..5]
function arrFunc(x, y) {
    let arr = [];
    for (let i = x; i <= y; i++) {
        arr.push(i);
    }
    return arr;
}

function applicatorFunc(inpFunc, s, x, y) {
    if (s == 's') {
        const arr = inpFunc(x, y);
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum;
    } else {
        const arr = inpFunc(x, y);
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum / (y - x + 1);
    }
}

let x = applicatorFunc(arrFunc, 's', 1, 10);
console.log(x);